MmSupervisorPkg/Drivers/MmPeiLaunchers/MmDxeSupport.c

```diff
@@ -771,6 +1045,33 @@ MmDxeSupportEntry (
     return Status;
   }
 
+  // MU_CHANGE: MM_SUPV: We are just making sure this communication to supervisor does not fail.
+  Status = UpdateDxeCommunicateBuffer (&VersionInfo, &NewCommBuffer);
+  if (EFI_ERROR (Status) && (Status != EFI_UNSUPPORTED)) {
+    DEBUG ((DEBUG_ERROR, "%a Failed to switch communication channel - %r\n", __FUNCTION__, Status));
+    return Status;
+  }
+
+  if (Status == EFI_UNSUPPORTED) {
+    // If it is not supported, we still publish the table using the hobs.

```

Comment:
In this case, a runtime data comm buffer does not exist, right?

Reply:
yeah, i was wondering if we should actually make it a hard requirement to update the comm buffer now, because the core private has to exist before the mm foundation is setup during PEI.

Thus pinging supervisor in DXE to see if this update is supported will not really be backwards compatible. Let me know if any thoughts.

Reply:
I think it is okay to make it a hard requirement and rev the version to indicate a breaking change.

Reply:
Thanks. I updated the logic to treat errors from this operation generally with asserts and bail. Verified on Q35 bootable to Windows. Will update the version to indicate the breaking change.

Reply:
Version is updated. Will make a new release once it is in.